### PR TITLE
Add workflow for testing Yosys plugin support

### PR DESCRIPTION
**Pull request by rkapuscik**
This adds a separate CI workflow that:
* checks out current confirmed-to-work tool versions from https://github.com/antmicro/yosys-uhdm-plugin-integration
* builds the tools against current Surelog
* uses the tools to build Ibex as a reasonably complex test case

This aims to help catch regressions that break the SystemVerilog plugin for Yosys.

### CODE REVIEW

**Review by kamilrakoczy — APPROVED** [review]
LGTM

**Comment by mithro** [comment]
Please make sure there is something like this that is checking the latest versions, so we know when an change causes breakage.

**Comment by mithro** [comment]
FYI - @hzeller

File: .github/workflows/yosys-plugin.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    steps:
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get update -qq
+        sudo apt install -y software-properties-common
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+        sudo apt-get update -qq
+        sudo apt install -y gcc-9 g++-9 build-essential cmake tclsh ant default-jre swig google-perftools libgoogle-perftools-dev python3 python3-dev python3-pip uuid uuid-dev tcl-dev flex libfl-dev git pkg-config libreadline-dev bison libffi-dev wget
```
Comment (hzeller) [inline .github/workflows/yosys-plugin.yml:31]: Surelog needs `python3-orderedmultidict` and `python3-psutil`, which are not mentioned here. Are they pulled in with `pip` ?

**Review by hzeller — APPROVED** [review]
... but double-check if we actually need to install an additional cmake.

File: .github/workflows/yosys-plugin.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    - name: Use ccache
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: linux-yosys-submodules
+
+    - name: Build binaries
+      run: |
+        cd yosys-uhdm-plugin-integration
+        # ubuntu bionic have too old cmake, install one from pip
+        pip install cmake
```
[the remark this answers is not in the document]
Comment (rkapuscik) [reply .github/workflows/yosys-plugin.yml]: Looks like Focal Fossa [has cmake 3.16.3](https://packages.ubuntu.com/focal/cmake) and Surelog [needs >=3.13](https://github.com/chipsalliance/Surelog/blob/master/INSTALL.md#build-dependencies). Removed.

File: .github/workflows/yosys-plugin.yml (hunk 1 above)
Comment (rkapuscik) [reply .github/workflows/yosys-plugin.yml:31]: `orderedmultidict` is installed through `pip` a few lines further. It seems that `psutil` is only needed for Surelog's regression tests and we're not running them here.

**Comment by alaindargelas** [comment]
@rkapuscik, This is still a work in progress looks like? It's crashing.

**Comment by hzeller** [comment]
@rkapuscik any update here ?

File: .github/workflows/yosys-plugin.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        submodules: recursive
+        fetch-depth: 0
+        path: 'yosys-uhdm-plugin-integration/Surelog'
+
+    - name: Setup cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.CCACHE_DIR }}
+        key: cache_${{ steps.cache_timestamp.outputs.time }}
+        restore-keys: cache_
```
Comment (kamilrakoczy) [inline .github/workflows/yosys-plugin.yml:60]: This requires ``cache_timestamp`` step, e.g.:
```
    - name: Create Cache Timestamp
      id: cache_timestamp
      uses: nanzm/get-time-action@v1.1
      with:
        format: 'YYYY-MM-DD-HH-mm-ss'
 ```

Comment (rkapuscik) [reply .github/workflows/yosys-plugin.yml:66]: Added, thanks!

**Review by kamilrakoczy — APPROVED** [review]

**Comment by rkapuscik** [comment]
@hzeller looks like it consistently passes now, so should be ready for review.

**Pull request merged**